postgres/9.6/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM postgres:9.6
+
+MAINTAINER Debezium Community
+
+# Install the packages which will be required to get everything to compile
+RUN apt-get update && apt-get install -y build-essential pkg-config git postgresql-server-dev-9.6 software-properties-common
+
+# The plugin requires protobuf-c at least 1.1 but the stable Debian repo has an older version, so we need to add the Debian testing repo which has the correct version
+RUN add-apt-repository "deb http://ftp.debian.org/debian testing main contrib" && apt-get update
+ 
+# Install the protobuf-c dependency for the plugin
+RUN apt-get install -y libprotobuf-c-dev=1.2.1-1+b1
+ 
+# Compile the plugin from sources and install it
+# TODO: the plugin should probably be forked in the DBZ repository
+RUN git clone https://github.com/hchiorean/decoderbufs
+RUN cd /decoderbufs && make && make install && cd /
```

Comment:
The Docker filesystem is immutable and is comprised of a number of layers (starting with the last immutable layer in the `FROM` image). When Dockerfiles are run, each step results in a new layer in the image. So, the one of the [best practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/) is to minimize the number of layers (and therefore size of the image) by minimizing the number commands.

And, each layer created by the Dockerfile contains the _net change_ of the Dockerfile command. Layers are never removed (they can be by other optimizing tools), so when one command clones a git repo, that layer will have all of the files in that repo. No matter what happens next, the image will contain that layer with the complete git repo.

But, since the layer is produced to hold the _net change_ of each `RUN` command, you can minimize the layer by using a single `RUN` command to do multiple things, and if the `RUN` command cleans up any intermediary but ultimately unnecessary files before it completes, those are left out of the layer.

Bottom line is that I'm suggesting replacing your two `RUN` commands with the following:

```
RUN git clone https://github.com/hchiorean/decoderbufs && \
    cd /decoderbufs && \
    make && \
    make install && \
    cd / && \
    rm -rf decoderbufs
```

The first few commands in `RUN` are the same, but after installing the library into the postgres installation and changing to the root directory, it also removes the directory that contains the git clone and any intermediary files created by `make` and `make install`. The _net change_ of this `RUN` command should thus be just the installed libraries in the postgres installation.